File: 2999-check-if-strings-can-be-made-equal-with-operations-i/2999-check-if-strings-can-be-made-equal-with-operations-i.cpp
class Solution {
public:
    bool canBeEqual(string s1, string s2) {
        for(int i=0;i<2;i++){
            if(s1[i] == s2[i] || s1[i] == s2[i+2] ){
                if(s1[i] == s2[i]) continue;
                else {
                    swap(s2[i],s2[i+2]);
                }
            }
            else{
                return false;
            }
        }
        
        if(s1[3] == s2[3] && s1[2] == s2[2])
            return true;
        return false;
    }
};

File: 2999-check-if-strings-can-be-made-equal-with-operations-i/2999-check-if-strings-can-be-made-equal-with-operations-i.py
class Solution:
    def canBeEqual(self, s1: str, s2: str) -> bool:
        for i in range(2):
            if s1[i] == s2[i] or s1[i] == s2[i + 2]:
                if s1[i] == s2[i]:
                    continue
                else:
                    s2_list = list(s2)
                    s2_list[i], s2_list[i + 2] = s2_list[i + 2], s2_list[i]
                    s2 = ''.join(s2_list)
            else:
                return False
        
        if s1[3] == s2[3] and s1[2] == s2[2]:
            return True
        return False